packages/rich-text/src/component/index.native.js

```diff
@@ -334,7 +333,6 @@ export class RichText extends Component {
 		if (
 			! onDelete ||
 			! isCollapsed( value ) ||
-			activeFormats.length ||
```

Comment:
On the web this checking the active formats makes sense because you
have a mode where you are inside a selection ("active selection").
That doesn't happen on mobile so having a selection active shouldn't
block the deletion and merge.

Reply:
Your explanation was helpful for understanding the fix. Nice one!